haystack/nodes/prompt/prompt_node.py

```diff
@@ -754,6 +754,7 @@ def prompt(self, prompt_template: Optional[Union[str, PromptTemplate]], *args, *
         :return: A list of strings as model responses.
         """
         results = []
+        prompt_collector: List[str] = kwargs.pop("prompt_collector", [])
```

Comment:
Hey @vblagoje and @zoltan-fedor could we consider adding `prompt_collector` to the output of `results` or have the function `prompt` return a tuple of `(results, prompt_collector)`? 

Right now a user must pass an external empty list called `prompt_collector` to the function `prompt` and then understand that the passed-in list is edited in place to be able to see the collected prompts. Overall, I don't think it's necessary to update/edit an input variable (`prompt_collector`) in this case and may be confusing to users. 

At the very least I think this behavior should be documented in the docstring of the `prompt` function.

Reply:
@srjl the prompt collector would be used only in pipelines. Non-pipeline users can turn on the logger and observe prompt logs. In some unusual edge cases where prompt logs are indeed needed for PromptNode alone(outside pipelines), they can pass that prompt_collector list.  I agree about the docs, spot on.

Reply:
Hey @vblagoje, @sjrl,
Would it be possible that the two of you figure out what you would like to have?

I have first submitted this PR with the `prompt_collector` being an instance variable - because just like @sjrl, I don't like the idea of a passed-in list being edited in place. But @vblagoje in his review of my PR asked me to change that to a passed-in list getting edited in place. Which then I did, but included it in the `prompt()` method's signature, so it can be documented. Which he didn't like and asked me to change to just use `kwargs` and pop it from there - which I did change, which is how we have arrived to where we are now.

Then comes the second level review from @sjrl, in which you ask me to not have an external list being passed in (which I only put it because I was asked to) and add documentation (which I have only removed because I have been asked to).

This double-review of directly conflicting directions is getting quite tiring. Could something be changed about this double-review process, so the PR's original author can be shielded from being pushed around in circles?

Thank you.

Reply:
Cool @zoltan-fedor we'll figure it out - thanks for your contribution.